### PR TITLE
SNAP-2110

### PR DESCRIPTION
**Pull request by kneeraj**
## Changes proposed in this pull request

Modified the log collection script to filter out unwanted lines from logs.
A list of patterns can be specified in the conf file and lines matching those patterns will not be collected by the script.

## Patch testing

Manual.

## ReleaseNotes.txt changes

None.

## Other PRs 

No.

### CODE REVIEW

File: cluster/sbin/collect-debug-artifacts.sh [hunk 1]
```diff
@@ -436,6 +439,20 @@ function collect_on_remote {
     fi
   done
 
+  # remove all unwanted lines from logs
+  if [ ! -z "${invert_pattern}" ] ; then
```
Comment (sumwale) [inline cluster/sbin/collect-debug-artifacts.sh]: Can be "-n" instead of "! -z"

File: cluster/sbin/collect-debug-artifacts.sh [hunk 2]
```diff
@@ -436,6 +439,20 @@ function collect_on_remote {
...
+    cd "${tmp_dir}"
+    for i in `ls *.log*`
+      do
+        invert_log_file_name=$i".inv"
```
Comment (sumwale) [inline cluster/sbin/collect-debug-artifacts.sh]: Double quote should be around whole to be effective: "$i.inv"

Though the for loop above will mess up any spaces in file names already. You will need something like:

```
for i in `ls *.log* | sed 's/^/"/;s/$/"/'`; do
```
to surround each filename in double quotes explicitly

**Comment by sumwale** [comment]
@kneeraj do you want to get this in for 1.0.1?

**Comment by PradeepSurale** [comment]
Can one of the admins verify this patch?

File: cluster/sbin/collect-debug-artifacts.sh (hunk 1 above)
Comment (bisenpiyush) [reply cluster/sbin/collect-debug-artifacts.sh]: Modified as suggested

**Pull request merged**